cloud/services/compute/vm.go

```diff
@@ -105,20 +109,60 @@ func (s *Service) CreateVm(machineScope *scope.MachineScope, spec *infrastructur
 
 	oscApiClient := s.scope.GetApi()
 	oscAuthClient := s.scope.GetAuth()
-	vmResponse, httpRes, err := oscApiClient.VmApi.CreateVms(oscAuthClient).CreateVmsRequest(vmOpt).Execute()
-	if err != nil {
-		fmt.Printf("Error with http result %s", httpRes.Status)
-		return nil, err
+	var vmResponse osc.CreateVmsResponse
+	createVmCallBack := func() (bool, error) {
+		var httpRes *_nethttp.Response
+		var err error
+		vmResponse, httpRes, err = oscApiClient.VmApi.CreateVms(oscAuthClient).CreateVmsRequest(vmOpt).Execute()
+		if err != nil {
+			if httpRes != nil {
+				fmt.Printf("Error with http result %s", httpRes.Status)
+			}
+			requestStr := fmt.Sprintf("%v", vmOpt)
+			allowedErrors := map[string]int{"AccessDenied": 503}
+			if reconciler.KeepRetryWithError(
+				requestStr,
+				httpRes,
+				err,
+				allowedErrors) {
+				return false, nil
+			}
+			return false, fmt.Errorf("could not create vm in Osc: %v", err)
+		}
+		return true, err
+	}
+	backoff := reconciler.EnvBackoff()
+	waitErr := wait.ExponentialBackoff(backoff, createVmCallBack)
+	if waitErr != nil {
+		return nil, waitErr
 	}
 	vms, ok := vmResponse.GetVmsOk()
 	if !ok {
 		return nil, errors.New("Can not get vm")
 	}
 	vmID := *(*vmResponse.Vms)[0].VmId
 	resourceIds := []string{vmID}
-	err = tag.AddTag("Name", vmName, resourceIds, oscApiClient, oscAuthClient)
-	if err != nil {
-		fmt.Printf("Error with http result %s", httpRes.Status)
+	addTagNameCallBack := func() (bool, error) {
+		err, httpRes := tag.AddTag("Name", vmName, resourceIds, oscApiClient, oscAuthClient)
+		if err != nil {
+			if httpRes != nil {
+				fmt.Printf("Error with http result %s", httpRes.Status)
+			}
+			allowedErrors := map[string]int{"AccessDenied": 503}
+			requestStr := fmt.Sprintf("Name")
```

Comment:
What  is it ? Your message will be 
```
Retry even if got (%v) error type with code error (%v) on request Name
```

You should show the request.

Reply:
Ok